ui/src/app/components/help-sidebar-icons.tsx

```diff
@@ -0,0 +1,667 @@
+              />
+            ),
+          ],
+          [
+            TerraJobStatus.FAILED,
+            () => (
+              <FontAwesomeIcon
+                icon={faCircle}
+                style={{
+                  ...styles.asyncOperationStatusIcon,
+                  color: colors.asyncOperationStatus.error,
+                }}
+              />
+            ),
+          ],
+          [
+            TerraJobStatus.SUCCEEDED,
+            () => (
+              <FontAwesomeIcon
+                icon={faCircle}
+                style={{
+                  ...styles.asyncOperationStatusIcon,
+                  color: colors.asyncOperationStatus.succeeded,
+                }}
+              />
+            ),
+          ],
+          [
+            TerraJobStatus.ABORTED,
+            () => (
+              <FontAwesomeIcon
+                icon={faCircle}
+                style={{
+                  ...styles.asyncOperationStatusIcon,
+                  color: colors.asyncOperationStatus.stopped,
+                }}
+              />
+            ),
+          ]
+        )}
+      </FlexRow>
+    </FlexRow>
+  );
+};
+
+interface DisplayIconProps {
+  workspace: WorkspaceData;
+  criteria: Array<Selection>;
+  concept?: Array<Criteria>;
+  runtimeStore: RuntimeStore;
+  compoundRuntimeOps: CompoundRuntimeOpStore;
+  genomicExtractionJobs: GenomicExtractionJob[];
+}
+const displayIcon = (icon: IconConfig, props: DisplayIconProps) => {
+  const {
+    workspace,
+    runtimeStore,
+    compoundRuntimeOps,
+    genomicExtractionJobs,
+    criteria,
+    concept,
+  } = props;
+  return switchCase(
+    icon.id,
+    [
+      'dataDictionary',
+      () => (
+        <a href={supportUrls.dataDictionary} target='_blank'>
+          <FontAwesomeIcon
+            data-test-id={'help-sidebar-icon-' + icon.id}
+            icon={icon.faIcon}
+            style={icon.style}
+          />
+        </a>
+      ),
+    ],
+    [
+      'runtime',
+      () =>
+        displayRuntimeIcon(icon, workspace, runtimeStore, compoundRuntimeOps),
+    ],
+    [
+      'terminal',
+      () => (
+        <RouteLink
+          path={`/workspaces/${workspace.namespace}/${workspace.id}/terminals`}
+        >
+          <FontAwesomeIcon
+            data-test-id={'help-sidebar-icon-' + icon.id}
+            icon={icon.faIcon}
+            style={icon.style}
+          />
+        </RouteLink>
+      ),
+    ],
+    [
+      'genomicExtractions',
+      () =>
+        displayExtractionIcon(icon, genomicExtractionJobs, criteria, concept),
+    ],
+    [
+      DEFAULT,
+      () =>
+        icon.faIcon === null ? (
+          <img
+            data-test-id={'help-sidebar-icon-' + icon.id}
+            src={proIcons[icon.id]}
+            style={icon.style}
+          />
+        ) : (
+          displayFontAwesomeIcon(icon, criteria, concept)
+        ),
+    ]
+  );
+};
+
+export const showCriteriaIcon = (pageKey: string, criteria: Array<Selection>) =>
```

Comment:
these 2 are new functions, exported because both components need them.  The originals are at help-sidebar.tsx L 334 and 344